### PR TITLE
rm pkg/workloads

### PR DESCRIPTION
**Pull request by aanm**
As we have deprecated the usage of any workload in v1.6 it's time to
remove the pkg/workloads entirely. The option flags will still be
available until v1.8.

Only one option was removed entirely which was --docker (or -e) as it
has been deprecated since v1.0.4.

Also, due this changes, besides the LOC removed, the cilium-agent binary
also decreased from 99MB -> 71M

Signed-off-by: André Martins <andre@cilium.io>

```release-note
Remove container runtime dependencies
```

Read per commit basis!

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9447)

### CODE REVIEW

**Review by brb — CHANGES_REQUESTED** [review]
This breaks ipvlan in cilium-docker (the libnetwork plugin). We should probably just remove the support for ipvlan from `plugins/cilium-docker/driver/driver.go`. /cc @borkmann

File: Documentation/install/upgrade.rst [hunk 1]
```diff
@@ -275,12 +275,21 @@ IMPORTANT: Changes required before upgrading to 1.7.0
     key-file: '/var/lib/cilium/etcd-client.key'
     cert-file: '/var/lib/cilium/etcd-client.crt'
 
+* Due the removal of external libraries to connect to container runtimes, Cilium
```
Comment (ianvernon) [inline Documentation/install/upgrade.rst]: "Due to the"

File: Documentation/install/upgrade.rst [hunk 2]
```diff
@@ -275,12 +275,21 @@ IMPORTANT: Changes required before upgrading to 1.7.0
...
+  is scheduled for v1.8.0
+
 Removed options
 ~~~~~~~~~~~~~~~~~~
 
 * ``lb``: The ``--lb`` feature has been removed. If you need load-balancing on
   a particular device, consider using :ref:`nodeport`.
 
+* ``docker`` and ``e``: This flags has been removed as Cilium no longer requires
+  container runtimes integrations to manage containers' networks.
```
Comment (ianvernon) [inline Documentation/install/upgrade.rst]: container runtime integrations

File: install/kubernetes/quick-install.yaml [hunk 3]
```diff
@@ -463,6 +454,7 @@ spec:
         - mountPath: /var/run/cilium
           name: cilium-run
       restartPolicy: Always
+      priorityClassName: system-node-critical
```
Comment (ianvernon) [inline install/kubernetes/quick-install.yaml]: I don't think this should be here - I recall that this got added when running `make` in the `install/kubernetes` directory when I was releasing a micro-release of `v1.6` and @tgraf said it shouldn't be there - @tgraf can you confirm?

**Review by ianvernon — COMMENTED** [review]
How does this work in the context of our `libnetwork` plugin overall? Is the only main impact the loss of the managing of existing container with Flannel?

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
never-tell-me-odds

**Comment by aanm** [comment]
never-tell-me-the-odds

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by ianvernon** [comment]
@aanm I fat-fingered and accidentally clicked on the button which created a merge-commit, sorry :( 

This was passing tests before upload; provided there are no more commits needed, we can merge without re-running tests.

File: Documentation/install/upgrade.rst [hunk 4]
```diff
@@ -277,12 +277,21 @@ IMPORTANT: Changes required before upgrading to 1.7.0
     key-file: '/var/lib/cilium/etcd-client.key'
     cert-file: '/var/lib/cilium/etcd-client.crt'
 
+* Due to the removal of external libraries to connect to container runtimes,
+  Cilium no longer supports the option ``flannel-manage-existing-containers``.
+  Cilium will still support integration with flannel for new containers
```
Comment (ianvernon) [inline Documentation/install/upgrade.rst]: casing: flannel --> Flannel

File: pkg/option/config.go [hunk 5]
```diff
@@ -110,9 +110,11 @@ const (
 	// ContainerRuntime sets the container runtime(s) used by Cilium
 	// { containerd | crio | docker | none | auto } ( "auto" uses the container
 	// runtime found in the order: "docker", "containerd", "crio" )
+	// Deprecated
```
Comment (ianvernon) [inline pkg/option/config.go]: Make this `Deprecated: <description>` c.f.: https://github.com/golang/go/wiki/Deprecated

File: pkg/option/config.go [hunk 6]
```diff
@@ -110,9 +110,11 @@ const (
...
 	ContainerRuntime = "container-runtime"
 
 	// ContainerRuntimeEndpoint set the container runtime(s) endpoint(s)
+	// Deprecated
```
Comment (ianvernon) [inline pkg/option/config.go]: Make this `Deprecated: <description>` c.f.: https://github.com/golang/go/wiki/Deprecated

File: pkg/option/config.go [hunk 7]
```diff
@@ -290,6 +292,7 @@ const (
 	// FlannelManageExistingContainers sets if Cilium should install the BPF
 	// programs on already running interfaces created by flannel. Require
 	// Cilium to be running in the hostPID.
+	// Deprecated
```
Comment (ianvernon) [inline pkg/option/config.go]: Make this `Deprecated: <description>` c.f.: https://github.com/golang/go/wiki/Deprecated

File: daemon/endpoint.go [hunk 8]
```diff
@@ -307,7 +307,11 @@ func (d *Daemon) createEndpoint(ctx context.Context, epTemplate *models.Endpoint
 }
 
 func (h *putEndpointID) Handle(params PutEndpointIDParams) middleware.Responder {
-	log.WithField(logfields.Params, logfields.Repr(params)).Debug("PUT /endpoint/{id} request")
+	if ep := params.Endpoint; ep != nil {
+		log.WithField("endpoint", logfields.Repr(*ep)).Debug("PUT /endpoint/{id} request")
```
Comment (ianvernon) [inline daemon/endpoint.go:311]: Non-blocking nit: Is there ever a case where we will care about the rest of the contents of `params` if `params.Endpoint` is not nil? I thought something like `%+v` would show all fields within `params` and not just a subset of them?

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

File: test/runtime/connectivity.go [hunk 9]
```diff
@@ -21,7 +35,6 @@ func runOnNetNextOnly(f func()) func() {
 	return func() {}
 }
 
-var _ = Describe("RuntimeConnectivityInVethModeTest", runtimeConnectivityTest("veth"))
```
Comment (ianvernon) [inline test/runtime/connectivity.go:24]: I don't fully understand this commit message - doesn't d391e52ee93777a031ec9337aa65b8d01eb2c63d add the way of plumbing the labels through to `cilium-agent`?

File: plugins/cilium-docker/driver/driver.go [hunk 10]
```diff
@@ -77,19 +82,28 @@ func newLibnetworkRoute(route route.Route) api.StaticRoute {
...
+	scopedLog := log.WithField("ciliumSockPath", ciliumSockPath)
+	c, err := client.NewClient(ciliumSockPath)
 	if err != nil {
 		scopedLog.WithError(err).Fatal("Error while starting cilium-client")
 	}
 
-	d := &driver{client: c}
+	scopedLog = scopedLog.WithField("dockerHostPath", dockerHostPath)
+	dockerCli, err := dockerCliAPI.NewClientWithOpts(
+		dockerCliAPI.WithVersion("v1.21"),
```
Comment (ianvernon) [inline plugins/cilium-docker/driver/driver.go:101]: Any specific need for this version? If so, hardcode it into a constant and state why it is used for self-documenting code.

File: pkg/endpoint/api.go [hunk 11]
```diff
@@ -509,8 +516,20 @@ func (e *Endpoint) ProcessChangeRequest(newEp *Endpoint, validPatchTransitionSta
...
+
+	if newEp.containerID != "" && e.containerID != newEp.containerID {
+		e.containerID = newEp.containerID
+	}
+
+	e.replaceInformationLabels(newEp.OpLabels.OrchestrationInfo)
+	rev := e.replaceIdentityLabels(newEp.OpLabels.IdentityLabels())
+	if rev != 0 {
+		// Run as a go routine since the runIdentityResolver needs to get the lock
+		go e.runIdentityResolver(context.Background(), rev, false)
```
Comment (ianvernon) [inline pkg/endpoint/api.go]: `context.Background()` --> `e.aliveCtx`?

**Review by ianvernon — COMMENTED** [review]
Looks great overall, but I do have a few nits / questions.

File: daemon/endpoint.go (hunk 8 above)
Comment (aanm) [reply daemon/endpoint.go:311]: The `ID`?

File: test/runtime/connectivity.go (hunk 9 above)
Comment (aanm) [reply test/runtime/connectivity.go:24]: Yes it does but it only works for containers with the network managed by libnetwork, not CNI.

File: plugins/cilium-docker/driver/driver.go (hunk 10 above)
Comment (aanm) [reply plugins/cilium-docker/driver/driver.go:101]: It's the same being used today. https://github.com/cilium/cilium/blob/cf7e154e73f4ca76faa28357753b7af6dfb282f8/pkg/workloads/docker.go#L103

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]
🎉 🎉 🎉

**Pull request merged**